### PR TITLE
small memory management clean-up init/ofi

### PR DESCRIPTION
**Pull request by kseager**
+included added error checking to environment variable conversion

### CODE REVIEW

**Comment by kseager** [comment]
OFI has an atomic max size we grab from atomicvalid.

in terms of the init assert, sure I can change it to an if statement, personally I think it would be nice to compile in two modes, one where you're developing a provider and need additionally safeguards, and one where you aren't, thus I liked the idea of an assert

**Comment by jdinan** [comment]
The checks that we typically compile out (in SOS they are actually opt-in with --enable-error-checking) are checks for whether or not the program uses the API correctly.  In this case, a correct program could run into something that we don't support properly.  I think we need to always have this error check, until it's fixed.

File: src/init.c [hunk 1]
```diff
@@ -67,6 +67,21 @@ static char shmem_internal_my_hostname[MAXHOSTNAMELEN];
...
+static inline long errchk_atol(char *s)
+{
+    long val;
+    char *e;
+    errno = 0;
+
+    val = strtol(s,&e,0);
+    if(errno != 0) {
+        perror("env var conversion");
+        exit(1);
```
Comment (jdinan) [inline src/init.c]: If the runtime is already initialized, we should use shmem_runtime_abort(...) here.

File: src/init.c [hunk 2]
```diff
@@ -67,6 +67,21 @@ static char shmem_internal_my_hostname[MAXHOSTNAMELEN];
...
+    }
```
Comment (jdinan) [inline src/init.c]: Unfortunately, the strtol error check doesn't do what you are hoping for.  See: http://stackoverflow.com/questions/9445171/strtol-using-errno

The better way to do this is to use sscanf:
```
int nread = sscanf(s, "%ld", &e);
if (nread != 1) {
  shmem_runtime_abort(1, "Error parsing environment variable");
}

File: src/init.c [hunk 3]
```diff
@@ -74,8 +89,13 @@ atol_scaled(char *s)
 {
     long val;
     char *e;
+    errno = 0;
 
     val = strtol(s,&e,0);
```
Comment (jdinan) [inline src/init.c]: Same comment as above, use sscanf instead of strtol.

File: src/init.c [hunk 4]
```diff
@@ -67,6 +67,21 @@ static char shmem_internal_my_hostname[MAXHOSTNAMELEN];
...
 
+
+static inline long errchk_atol(char *s)
+{
+    long val;
+    char *e;
+    errno = 0;
+
+    val = strtol(s,&e,0);
+    if(errno != 0) {
```
Comment (jdinan) [inline src/init.c]: Need to also check for "e == s" which indicates a parsing error.

**Comment by jdinan** [comment]
Thanks!

**Pull request merged**